test/e2e_node/container_lifecycle_test.go

```diff
@@ -948,6 +948,209 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
 			})
 		})
 	})
+
+	ginkgo.When("Running a pod with init containers and regular containers, restartPolicy=Always", func() {
+		ginkgo.When("the init container is updated with a new image", func() {
+			init1 := "init-1"
+			regular1 := "regular-1"
+			updatedImage := busyboxImage
+
+			originalPodSpec := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "initcontainer-update-img",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyAlways,
+					InitContainers: []v1.Container{
+						{
+							Name:  init1,
+							Image: agnhostImage,
+							Command: ExecCommand(init1, execCommand{
+								Delay:    1,
+								ExitCode: 0,
+							}),
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:  regular1,
+							Image: agnhostImage,
+							Command: ExecCommand(regular1, execCommand{
+								Delay:    20,
+								ExitCode: 0,
+							}),
+						},
+					},
+				},
+			}
+			preparePod(originalPodSpec)
+
+			ginkgo.It("should not restart init container when updated with a new image after finishing", func(ctx context.Context) {
+				client := e2epod.NewPodClient(f)
+				podSpec := client.Create(ctx, originalPodSpec)
+
+				ginkgo.By("running the pod", func() {
+					err := e2epod.WaitForPodNameRunningInNamespace(ctx, f.ClientSet, podSpec.Name, podSpec.Namespace)
+					framework.ExpectNoError(err)
+				})
+
+				ginkgo.By("updating the image", func() {
+					client.Update(ctx, podSpec.Name, func(pod *v1.Pod) {
+						pod.Spec.InitContainers[0].Image = updatedImage
+					})
+				})
+
+				ginkgo.By("verifying that the containers do not restart", func() {
+					podSpec, err := client.Get(ctx, podSpec.Name, metav1.GetOptions{})
+					framework.ExpectNoError(err)
+
+					err = e2epod.WaitForPodCondition(ctx, f.ClientSet, podSpec.Namespace, podSpec.Name, "the init container terminated regularly", 30*time.Second, func(pod *v1.Pod) (bool, error) {
```

Comment:
Does this verify nothing has been changed in the container status? I think this verification succeeds immediately. Wouldn't it be better to use `Consistently`?